File: C/AquecimentoIterativo.c
/*
   ==UserScript==
 @name         LAB01Q03 - Aquecimento Iterativo
 @namespace    https://github.com/Ddavidi/VERDE-PUC-MINAS
 @description  VERDE PUC MINAS - Aquecimento Iterativo
 @author       @ddavidi_
   ==/UserScript==
*/

#include <stdio.h>
#include <string.h>


void contarMaiusculas(char entrada[]) {
    int tamanho = strlen(entrada);
    int coutMaiusculas = 0;

    for(int i=0; i<tamanho; i++) {
        if(entrada[i] >= 65 && entrada[i] <= 90)
            coutMaiusculas++;
    }

    printf("%d\n", coutMaiusculas);
}

int main (void) {
    char entrada[500];

    fgets(entrada, sizeof(entrada), stdin);

    while(strcmp(entrada, "FIM\n") != 0) {
        contarMaiusculas(entrada);
        fgets(entrada, sizeof(entrada), stdin);
    }

    return 0;
}

File: JAVA/AquecimentoIterativo.java
/*
   ==UserScript==
 @name         LAB01Q01 - Aquecimento Iterativo
 @namespace    https://github.com/Ddavidi/VERDE-PUC-MINAS
 @description  VERDE PUC MINAS - Aquecimento Iterativo
 @author       @ddavidi_
   ==/UserScript==
*/

import java.util.Scanner;

public class AquecimentoIterativo {

    public static void contarMaiusculas(String entrada){
        int tamanho = entrada.length();
        int coutMaiusculas = 0;

        for(int i=0; i<tamanho; i++){
            if (entrada.charAt(i) >= 65 && entrada.charAt(i) <= 90)
                coutMaiusculas++;
        }

        System.out.println(coutMaiusculas);

    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String entrada = scan.nextLine();
        
        while (!entrada.equals("FIM")) {
            contarMaiusculas(entrada);
            entrada = scan.nextLine();
        }

        scan.close();
    }
}